config/socket/game.js

```diff
@@ -1,428 +1,577 @@
-        var tableCard = [];
-        for (var i = 0; i < thisCardArray.length; i++ ) {
-          var cardIndex = null;
-          for (var j = 0; j < this.players[playerIndex].hand.length; j++) {
-            if (this.players[playerIndex].hand[j].id === thisCardArray[i]) {
-              cardIndex = j;
+        let previouslySubmitted = false;
+        _.each(this.table, (pickedSet, index) => {
+          if (pickedSet.player === thisPlayer) {
+            previouslySubmitted = true;
+          }
+        });
+        if (!previouslySubmitted) {
+        // Find the indices of the cards in the player's
+        // hand (given the card ids)
+          const tableCard = [];
+          for (let i = 0; i < thisCardArray.length; i += 1) {
+            let cardIndex = null;
+            for (let j = 0; j < this.players[playerIndex].hand.length; j += 1) {
+              if (this.players[playerIndex].hand[j].id === thisCardArray[i]) {
+                cardIndex = j;
+              }
             }
+            console.log('card', i, 'is at index', cardIndex);
+            if (cardIndex !== null) {
+              tableCard.push(this.players[playerIndex].hand.splice(cardIndex, 1)[0]);
+            }
+            console.log('table object at', cardIndex, ':', tableCard);
           }
-          console.log('card',i,'is at index',cardIndex);
-          if (cardIndex !== null) {
-            tableCard.push(this.players[playerIndex].hand.splice(cardIndex,1)[0]);
+          if (tableCard.length === this.curQuestion.numAnswers) {
+            this.table.push({
+              card: tableCard,
+              player: this.players[playerIndex].socket.id
+            });
+          }
+          console.log('final table object', this.table);
+          if (this.table.length === this.players.length - 1) {
+            clearTimeout(this.choosingTimeout);
+            this.stateJudging(this);
+          } else {
+            this.sendUpdate();
           }
-          console.log('table object at',cardIndex,':',tableCard);
-        }
-        if (tableCard.length === this.curQuestion.numAnswers) {
-          this.table.push({
-            card: tableCard,
-            player: this.players[playerIndex].socket.id
-          });
-        }
-        console.log('final table object',this.table);
-        if (this.table.length === this.players.length-1) {
-          clearTimeout(this.choosingTimeout);
-          this.stateJudging(this);
-        } else {
-          this.sendUpdate();
         }
       }
+    } else {
+      console.log('NOTE:', thisPlayer, 'picked a card during', this.state);
     }
-  } else {
-    console.log('NOTE:',thisPlayer,'picked a card during',this.state);
   }
-};
 
-Game.prototype.getPlayer = function(thisPlayer) {
-  var playerIndex = this._findPlayerIndexBySocket(thisPlayer);
-  if (playerIndex > -1) {
-    return this.players[playerIndex];
-  } else {
+  /**
+   * 
+   * @param {*} thisPlayer
```

Comment:
What does * mean?

Reply:
It means the method can take parameter of any data type which is not the case in this context, It was initially included because my code editor automatically inserted it there, I left it because I didn't know what should go there. due changes will be effected immediately.